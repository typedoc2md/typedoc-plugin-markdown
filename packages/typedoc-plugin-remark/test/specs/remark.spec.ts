import * as fs from 'fs';
import * as path from 'path';

describe(`Remark`, () => {
  describe(`Simple`, () => {
    test(`should parse modules page`, async () => {
      const contents = fs
        .readFileSync(path.join(__dirname, '../out/md/simple/modules.md'))
        .toString();
      expect(contents).toMatchSnapshot();
    });
    test(`should parse documents page`, async () => {
      const contents = fs
        .readFileSync(
          path.join(__dirname, '../out/md/simple/documents/toc-doc.md'),
        )
        .toString();
      expect(contents).toMatchSnapshot();
    });
  });

  describe(`Conditional`, () => {
    test(`should parse README page kind`, async () => {
      const contents = fs
        .readFileSync(path.join(__dirname, '../out/md/conditional/README.md'))
        .toString();
      expect(contents).toMatchSnapshot();
    });

    test(`should parse Index page kind`, async () => {
      const contents = fs
        .readFileSync(path.join(__dirname, '../out/md/conditional/modules.md'))
        .toString();
      expect(contents).toMatchSnapshot();
    });

    test(`should parse documents page`, async () => {
      const contents = fs
        .readFileSync(
          path.join(__dirname, '../out/md/simple/documents/toc-doc.md'),
        )
        .toString();
      expect(contents).toMatchSnapshot();
    });

    test(`should parse class page`, async () => {
      const contents = fs
        .readFileSync(
          path.join(__dirname, '../out/md/simple/remark/classes/Class.md'),
        )
        .toString();
      expect(contents).toMatchSnapshot();
    });

    test(`should parse interface page`, async () => {
      const contents = fs
        .readFileSync(
          path.join(
            __dirname,
            '../out/md/simple/remark/interfaces/Interface.md',
          ),
        )
        .toString();
      expect(contents).toMatchSnapshot();
    });
  });
});
