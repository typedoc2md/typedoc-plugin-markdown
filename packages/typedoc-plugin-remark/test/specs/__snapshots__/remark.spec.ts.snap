// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remark Conditional should parse Index page kind 1`] = `
"---
title: "test"
---

# typedoc-plugin-remark

## Documents

+ [toc-doc](documents/toc-doc.md)
+ [toc-doc-2](documents/toc-doc-2.md)

## Modules

+ [remark](remark/README.md)

> thanks [**@tgreyuk**](https://github.com/tgreyuk)

## New Index Heading
"
`;

exports[`Remark Conditional should parse README page kind 1`] = `
"# Project README

Some text

## Some heading 2

Some text

+ list item 1
+ list item 2

## Another heading 2

thanks [**@tgreyuk**](https://github.com/tgreyuk)

\`\`\`ts
const x = 1;
\`\`\`

## New Readme Heading
"
`;

exports[`Remark Conditional should parse class page 1`] = `
"---
title: "test"
---

# Class: Class

## Constructors

### Constructor

> **new Class**(): \`Class\`

#### Returns

\`Class\`

## Properties

### a

> **a**: \`string\`

***

### b

> **b**: \`number\`

> thanks [**@tgreyuk**](https://github.com/tgreyuk)
"
`;

exports[`Remark Conditional should parse documents page 1`] = `
"# TOC Document

Some text

## Some heading 2

Some text

## Another heading 2

thanks [**@tgreyuk**](https://github.com/tgreyuk)
"
`;

exports[`Remark Conditional should parse interface page 1`] = `
"---
title: "test"
---

# Interface: Interface

## Properties

### a

> **a**: \`string\`

***

### b

> **b**: \`number\`

> thanks [**@tgreyuk**](https://github.com/tgreyuk)
"
`;

exports[`Remark Simple should parse documents page 1`] = `
"# TOC Document

Some text

## Some heading 2

Some text

## Another heading 2

thanks [**@tgreyuk**](https://github.com/tgreyuk)
"
`;

exports[`Remark Simple should parse modules page 1`] = `
"---
title: "test"
---

# typedoc-plugin-remark

## Documents

* [toc-doc](documents/toc-doc.md)
* [toc-doc-2](documents/toc-doc-2.md)

## Modules

* [remark](remark/README.md)

> thanks [**@tgreyuk**](https://github.com/tgreyuk)
"
`;
